cypher.y

```diff
@@ -37,137 +37,148 @@ int order_clause_direction = 1; // 1 for ascending, -1 for descending
     ;
 
 map_entry:
-    IDENTIFIER COLON expression
+    IDENTIFIER COLON expression { printf("Map entry parsed successfully.\n"); }
     ;
 
 expression:
-    INTEGER
-    | STRING
-    | IDENTIFIER
+    INTEGER { printf("Integer expression parsed: %d.\n", $1); }
+    | STRING { printf("String expression parsed: %s.\n", $1); }
+    | IDENTIFIER { printf("Identifier expression parsed: %s.\n", $1); }
     ;
 
-where_clause_opt:
-    /* empty */
-    | WHERE expression
+where_clause:
+    | WHERE expression { printf("WHERE clause parsed successfully.\n"); }
     ;
 
-with_clause_opt:
-    /* empty */
-    | WITH expression_list return_clause
+with_clause:
+    WITH expression_list { printf("WITH clause parsed successfully.\n"); }
     ;
 
 expression_list:
-    expression
-    | expression_list COMMA expression
-    ;
+	expression { printf("Expression parsed successfully.\n"); }
+	| expression_list COMMA expression { printf("Expression parsed successfully.\n"); }
+	;
 
 return_clause:
-    RETURN return_item_list order_clause_opt
+	RETURN return_item_list order_clause_opt skip_clause_opt limit_clause_opt { printf("Return clause parsed successfully.\n"); }
 ;
 
 return_item_list:
-    return_item
-    | return_item_list COMMA return_item
-    ;
+	return_item { printf("Return item parsed successfully.\n"); }
+	| return_item_list COMMA return_item { printf("Return item parsed successfully.\n"); }
+	;
 
 return_item:
-    expression
-    | expression AS IDENTIFIER
-    ;
+	expression { printf("Return item parsed successfully.\n"); }
+	| expression AS IDENTIFIER { printf("Return item with alias parsed successfully.\n"); }
+	;
 
 order_clause_opt:
-    /* empty */
-    | ORDER BY sort_item_list
-    ;
+	/* empty */ { printf("No ORDER BY clause.\n"); }
+	| ORDER BY sort_item_list { printf("ORDER BY clause parsed successfully.\n"); }
+	;
 
 sort_item_list:
-    sort_item
-    | sort_item_list COMMA sort_item
-    ;
+	sort_item { printf("Sort item parsed successfully.\n"); }
+	| sort_item_list COMMA sort_item { printf("Sort item parsed successfully.\n"); }
+	;
 
 sort_item:
-    expression sort_direction_opt
-    ;
+	expression sort_direction_opt { printf("Sort item parsed successfully.\n"); }
+	;
 
 sort_direction_opt:
-    /* empty */
-    | ASC
-    | DESC
-    ;
+	/* empty */ { printf("Sort direction not specified; defaulting to ASC.\n"); order_clause_direction = 1; }
+	| ASC { printf("Sort direction specified: ASC.\n"); order_clause_direction = 1; }
+	| DESC { printf("Sort direction specified: DESC.\n"); order_clause_direction = -1; }
+	;
+
+skip_clause_opt:
+	/* empty */ { printf("No SKIP clause.\n"); }
+	| SKIP INTEGER { printf("SKIP clause parsed: %d.\n", $2); }
+	;
+
+limit_clause_opt:
+	/* empty */ { printf("No LIMIT clause.\n"); }
+	| LIMIT INTEGER { printf("LIMIT clause parsed: %d.\n", $2); }
+	;
 
 %%
 
 void yyerror(char const *s)
 {
-    fprintf(stderr, "Parser error: %s\n", s);
+	printf("Parser error: %s\n", s);
```

Comment:
Change message,should look like postgresql error

Reply:
Fixed!